sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -603,7 +603,14 @@ case class ExternalMapToCatalyst private(
 
   override def foldable: Boolean = false
 
-  override def dataType: MapType = MapType(keyConverter.dataType, valueConverter.dataType)
+  override def dataType: MapType = {
+    val isPrimitiveType = valueType match {
+      case BooleanType | ByteType | ShortType | IntegerType | LongType |
+            FloatType | DoubleType => true
+      case _ => false
+    }
+    MapType(keyConverter.dataType, valueConverter.dataType, !isPrimitiveType)
```

Comment:
@cloud-fan I'm afraid I don't think it's a good idea..

There are 3 cases that we use `LambdaVariable`:

- As a placeholder of element for `MapObjects`.
- As a placeholder of key for `ExternalMapToCatalyst`
- As a placeholder of value for `ExternalMapToCatalyst`

Only 1 case we can fix nullability by it is the second case and it would not be the cause of this problem.